code/_onclick/hud/fullscreen.dm

```diff
@@ -7,15 +7,28 @@
 		screen = screens[category]
 		if(screen.type != type)
 			clear_fullscreen(category, FALSE)
-			return .()
```

Comment:
It SEEMS TO ME that the code here clears the fullscreen *then runs the proc again* to continue in case you're trying to override an overlay category that's already present.